book/src/pieces/intro/single_line_if_else.md

```diff
@@ -0,0 +1,84 @@
+# Single line if-else
+
+Both Python and Rust support single line if-else statements. This is especially useful when
+performing simple operations on a value, allowing for more concise code.
+
+## Python
+
+Consider the following function in which we print a message depending on whether a person is
+born in a leap year or not.
+
+To do this, we first define a function `approx_year_of_birth` that returns the approximate year.
+
+```py
+def approx_year_of_birth(person: Person) -> int:
+    birth_year_approx = datetime.now().year - person.age
+    return birth_year_approx
+```
+
+We then use this function after initializing a list of `Person` objects.
+
+```py
+def run6() -> None:
+    persons = [Person("Josephine", 20), Person("Wesley", 31)]
+    for person in persons:
+        # Check if person is born in a leap year using simplistic leap year logic
+        birth_year = approx_year_of_birth(person)
+        person_is_born_in_leap_year = True if birth_year % 4 == 0 else False
```

Comment:
Calculating a leap year is actually more complicated than this, but this is probably OK for illustration purposes here. Maybe just mention this so we don't get PRs trying to fix it. Or I guess that is what the comment is already doing?

1. If the year is evenly divisible by 4, go to step 2. Otherwise, go to step 5.
2. If the year is evenly divisible by 100, go to step 3. Otherwise, go to step 4.
3. If the year is evenly divisible by 400, go to step 4. Otherwise, go to step 5.
4. The year is a leap year (it has 366 days).
5. The year is not a leap year (it has 365 days).

Reply:
Yes, the comment is doing exactly that and the function name is clearly saying "approx". I of course didn't want to write a correct leap year calculator because that would make it way too involved for the purposes of illustration 😅

Reply:
But you're right, I wonder if we're going to see future PRs trying to fix exactly this. Programmers are known to be a pedantic bunch at times 🤣.